Sources/Controllers/Map/Widgets/OATopTextView.mm

```diff
@@ -548,6 +548,23 @@ - (BOOL) updateInfo
         }
     }
     
+    //Change several shields to one
+    if (streetName.shields.count > 1)
```

Comment:
According to @vshcherb comment
Problem described in [the comment to issue](https://github.com/osmandapp/OsmAnd/issues/21362#issuecomment-2493801878)